src/types.ts

```diff
@@ -0,0 +1,54 @@
+import type { CSSProperties } from 'react';
+import { breakpoint } from '@core-ds/primitives';
+
+export type Breakpoint = typeof breakpoint;
+
+export type ResponsiveStyle<Value> = {
+   [B in keyof Breakpoint]: Value;
+} & { base?: Value };
+
+export type Style<Value> = ResponsiveStyle<Value> | Value;
+
+export interface LayoutProps {
+   margin?: Style<CSSProperties['margin']>;
+   marginX?: Style<CSSProperties['margin']>;
```

Comment:
It seems like every time we use `Style` we have to pass a CSSProperties value to it. Would it make sense to have Style take the key of CSSProperties instead?

```ts
export type Style<Property extends keyof CSSProperties> =
   ResponsiveStyle<CSSProperties[Property]> | CSSProperties[Property];

export interface LayoutProps {
   margin?: Style<'margin'>,
   ...
}

Reply:
Also wouldn't marginX want to use the marginLeft type? I don't think `marginX="1px 2px 3px 4px"` would make sense.

Reply:
> It seems like every time we use `Style` we have to pass a CSSProperties value to it. Would it make sense to have Style take the key of CSSProperties instead?
> 
> ```ts
> export type Style<Property extends keyof CSSProperties> =
>    ResponsiveStyle<CSSProperties[Property]> | CSSProperties[Property];
> 
> export interface LayoutProps {
>    margin?: Style<'margin'>,
>    ...
> }
> ```

It's a nice suggestion. How would you deal with [this](https://github.com/iFixit/footer/blob/f5172c2432eb2407b01e3822836297ca6346a3a5/src/Stack/index.tsx#L16-L16) though? Would you use `margin-left`? 



> Also wouldn't marginX want to use the marginLeft type? I don't think `marginX="1px 2px 3px 4px"` would make sense.

Agree 👍 
```suggestion
export type Style<Value> = ResponsiveStyle<Value> | Value;

export interface LayoutProps {
   margin?: Style<CSSProperties['margin']>;
   marginX?: Style<CSSProperties['margin-left']>;
```